src/Spice86.Core/Emulator/InterruptHandlers/Dos/Xms/ExtendedMemoryManager.cs

```diff
@@ -0,0 +1,490 @@
+        int handle = _state.DX;
+
+        if (!_xmsHandles.TryGetValue(handle, out int lockCount)) {
+            _state.AX = 0; // Didn't work.
+            _state.BL = 0xA2; // Invalid handle.
+            return;
+        }
+
+        if (lockCount > 0) {
+            _state.AX = 0; // Didn't work.
+            _state.BL = 0xAB; // Handle is locked.
+            return;
+        }
+
+        if (TryGetBlock(handle, out XmsBlock? block)) {
+            XmsBlock freeBlock = block.Free();
+            _xmsBlocksLinkedList.Replace(block, freeBlock);
+            MergeFreeBlocks(freeBlock);
+        }
+
+        _xmsHandles.Remove(handle);
+        _state.AX = 1; // Success.
+    }
+
+    /// <summary>
+    /// Locks a block of memory.
+    /// </summary>
+    public void LockExtendedMemoryBlock() {
+        int handle = _state.DX;
+
+        if (!_xmsHandles.TryGetValue(handle, out int lockCount)) {
+            _state.AX = 0; // Didn't work.
+            _state.BL = 0xA2; // Invalid handle.
+            return;
+        }
+
+        _xmsHandles[handle] = lockCount + 1;
+
+        _ = TryGetBlock(handle, out XmsBlock? block);
+        uint fullAddress = XmsBaseAddress + block?.Offset ?? 0;
+
+        _state.AX = 1; // Success.
+        _state.DX = (ushort)(fullAddress >> 16);
+        _state.BX = (ushort)(fullAddress & 0xFFFFu);
+    }
+
+    /// <summary>
+    /// Unlocks a block of memory.
+    /// </summary>
+    public void UnlockExtendedMemoryBlock() {
+        int handle = _state.DX;
+
+        if (!_xmsHandles.TryGetValue(handle, out int lockCount)) {
+            _state.AX = 0; // Didn't work.
+            _state.BL = 0xA2; // Invalid handle.
+            return;
+        }
+
+        if (lockCount < 1) {
+            _state.AX = 0;
+            _state.BL = 0xAA; // Handle is not locked.
+            return;
+        }
+
+        _xmsHandles[handle] = lockCount - 1;
+
+        _state.AX = 1; // Success.
+    }
+
+    /// <summary>
+    /// Returns information about an XMS handle.
+    /// </summary>
+    public void GetHandleInformation() {
+        int handle = _state.DX;
+
+        if (!_xmsHandles.TryGetValue(handle, out int lockCount)) {
+            _state.AX = 0; // Didn't work.
+            _state.BL = 0xA2; // Invalid handle.
+            return;
+        }
+
+        _state.BH = (byte)lockCount;
+        _state.BL = (byte)(MaxHandles - _xmsHandles.Count);
+
+        if (!TryGetBlock(handle, out XmsBlock? block)) {
+            _state.DX = 0;
+        } else {
+            _state.DX = (ushort)(block.Length / 1024u);
+        }
+
+        _state.AX = 1; // Success.
+    }
+
+    /// <summary>
+    /// Copies a block of memory.
+    /// TODO: Verify this works
+    /// </summary>
+    public unsafe void MoveExtendedMemoryBlock() {
+        bool a20State = _memory.A20Gate.IsEnabled;
+        _memory.A20Gate.IsEnabled = true;
+
+        XmsMoveData moveData = new XmsMoveData(_memory, _state.DS);
```

Comment:
Has this been executed by a real program? I am asking because XmsMoveData is not reading from emulated memory as far as I can see so data will be all 0